curator-test/src/main/java/org/apache/curator/test/InstanceSpec.java

```diff
@@ -205,6 +205,10 @@ public int getQuorumPort()
         return quorumPort;
     }
 
+    /**
+     * @deprecated use {@link TestingServer#getConnectString()} or {@link TestingCluster#getConnectString()} instead
+     */
+    @Deprecated
```

Comment:
If we already remove `public QuorumPeerConfig getConfig() throws Exception {` which breaks compatibility, what if we remove this deprecated mark and change the visibility to default (pacakge)?

Reply:
... and why do we remove `ZooKeeperMainFace#getConfig`?

Reply:
Actually, this pr mixed two things: CURATOR-535 and bring back `TestTestingServer#setCustomTickTimeTest` for #383.

`ZooKeeperMainFace#getConfig` was introduced in #434, and its sole external usage in `TestTestingServer#setCustomTickTimeTest` undermines #383.

Comparing to public `InstanceSpec`, I think `ZooKeeperMainFace` is more like an internal concept. May be we can restrict visibility of `ZooKeeperMainFace` to package ? This should solve all above concerns if make sense to you.

Reply:
@kezhuw That sounds reasonable. And we can add `getTickTime` or something so that we don't have to change `hasZooKeeperServerEmbedded`.

Reply:
I have removed `public` from `ZooKeeperMainFace`. @tisonkun 

`getTickTime` needs reflection as `ZooKeeperServerEmbedded` exposes no such info([ZOOKEEPER-4670](https://issues.apache.org/jira/browse/ZOOKEEPER-4670)). In previous version(8eda7d0f73333a5e0781d38006065eac98a012a4), I exposed a get method in `ZooKeeperMainFace` to retrieve `ServerCnxnFactory` where both `getTickTime` and `getClientPort` could be easily constructed . But that approach depends heavily on reflection as `QuorumPeerMain` and `ZooKeeperServerEmbedded` exposes no such info.